vyper/builtins/_convert.py

```diff
@@ -463,7 +463,7 @@ def to_flag(expr, arg, out_typ):
 def convert(expr, context):
     assert len(expr.args) == 2, "bad typecheck: convert"
 
-    arg_ast = expr.args[0]
+    arg_ast = expr.args[0].reduced()
```

Comment:
What is a good way to test this? Should I check that the bounds check is not in the IR?

Reply:
i think you can check that some examples get rejected at compile time (like the examples @ritzdorf gave). but i think for `convert()` it's actually not that critical, since there will just be a runtime error. the `slice()` error is more important since it blocks some user workflow.